docs/proposals/kube_events_design_doc.md

```diff
@@ -0,0 +1,137 @@
+Collect and store kubernetes events
+===
+By kubernetes events we understand log messages internal to kubernetes, accessible through the kubernetes API `/api/v1/events?watch=true`, originally stored in etcd. The etcd storage has time and performance constraints, therefore, we would like to collect and store them permanently in EFK.
+
+How are the logs collected
+---
+By default, fluentd is able to read any pod’s STDOUT and insert as an enriched log message to Elasticsearch. The eventrouter outputs the events as a JSON string, parseable by fluentd which sets the event related fields in the `_default_` namespace. The desired structure is achieved using a [fluentd filter record_transformer](https://github.com/openshift/origin-aggregated-logging/pull/573).
+
+Selected implementation consists of:
+- [eventrouter](https://github.com/openshift/openshift-ansible/pull/4973) - scraping the events from kubernetes API and printing as separate messages to STDOUT for further processing.
+- [fluentd record_transformer](https://github.com/openshift/origin-aggregated-logging/pull/573) - transforming the log message to adhere to desired data model moving event related fields to nest under the `kubernetes` field.
+    
+![design_proposal_image](https://cdn.rawgit.com/wozniakjan/origin-aggregated-logging/37bb71012f7a0008b929bb88352f78ad023dbe15/docs/proposals/kube_event_collection.svg)
+
+What do the event logs contain
+---
+The logs contain all of the standard fields and metadata as described in [ViaQ CDM](https://github.com/ViaQ/elasticsearch-templates/tree/master/namespaces). Additionally, there are following fields from the event nested under the `kubernetes` field for deduplication:
+- message <- event.message
+- kubernetes.event.source_component <- event.source.component
+
+In case the event was triggered by a pod, additional deduplication occurs:
+- kubernetes.pod_name <- event.involvedObject.name
+- kubernetes.namespace_name <- event.involvedObject.namespace
+- kubernetes.pod_id is deleted
+- kubernetes.namespace_id is deleted
```

Comment:
why are pod_id and namespace_id deleted?  note that these fields are required for storage in a `project.namespace.*` index rather than the `.operations.*` index.